src/Serilog/Context/LogContext.cs

```diff
@@ -88,15 +89,59 @@ public static IDisposable Push(ILogEventEnricher enricher)
     /// <param name="enrichers">Enrichers to push onto the log context</param>
     /// <returns>A token that must be disposed, in order, to pop properties back off the stack.</returns>
     /// <exception cref="ArgumentNullException">When <paramref name="enrichers"/> is <code>null</code></exception>
+    [OverloadResolutionPriority(0)]
     public static IDisposable Push(params ILogEventEnricher[] enrichers)
+    {
+        return Push(enrichers);
```

Comment:
If this ends up resolving as `IEnumerable` there will be a perf regression; the earlier version used numeric indexing because there's no struct enumerator for `IEnumerable`. Possible to keep `AsSpan` here to sort this out, plus make the target method easier to identify?

Reply:
If `AsSpan` is used, a guard against null arg will be needed.

Reply:
> If AsSpan is used, a guard against null arg will be needed.

Added this, LGTM, will merge if my changes come through green :-)